OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -128,4 +128,17 @@ void Frame::extendRealizeTopology(SimTK::State& s) const
 {
     Super::extendRealizeTopology(s);
     groundTransformIndex = getCacheVariableIndex("ground_transform");
+}
+void Frame::extendRealizeVelocity(const SimTK::State& s) const
+{   
+    //Evaluate the lazy cache for the groundTransform so it doesn't get
+    //needlesly calculated by each one of the threads while realizing
+    //dynamics. (Also for thread-safety reasons)
+    Super::extendRealizeVelocity(s);
+
+    SimTK::Value<SimTK::Transform>::downcast(getSystem().getDefaultSubsystem().
+        updCacheEntry(s, groundTransformIndex)).upd() = calcGroundTransform(s);
+    
+    getSystem().getDefaultSubsystem().
+        markCacheValueRealized(s, groundTransformIndex);
```

Comment:
This is one of the issues I ran into - when moving it up to Position, for some reason some forces would not pass such as Force::Gravity.